src/Export/Search/ProductCriteriaBuilder.php

```diff
@@ -349,4 +378,11 @@ protected function getVisibilityFilterForRealVariants(): ProductAvailableFilter
             ProductVisibilityDefinition::VISIBILITY_SEARCH
         );
     }
+
+    public function withAdvancedPricing(): self
+    {
+        $this->criteria->addAssociation('prices.rule.conditions');
+
+        return $this;
+    }
```

Comment:
Is this association still needed?

Reply:
Yep, without it we can't get advanced prices.